util/src/main/java/io/kubernetes/client/util/ModelMapper.java

```diff
@@ -114,33 +147,35 @@ public static void addModelMap(String group, String version, String kind, Class<
    * @return the api type class
    */
   public static Class<?> getApiTypeClass(String group, String version, String kind) {
-    Class<?> clazz = classes.get(new GroupVersionKind(group, version, kind));
+    Class<?> clazz = classesByGVK.get(new GroupVersionKind(group, version, kind));
     if (clazz != null) {
       return clazz;
     }
-    return classes.get(new GroupVersionKind(null, version, kind));
+    return preBuiltGetApiTypeClass(group, version, kind);
   }
 
   /**
-   * Returns all the registered model classes.
+   * Gets the GVK by the given model class.
    *
-   * @return the all known classes
+   * @param clazz the clazz
+   * @return the group version kind by class
    */
-  public static Map<GroupVersionKind, Class<?>> getAllKnownClasses() {
-    return ImmutableMap.copyOf(classes);
+  public static GroupVersionKind getGroupVersionKindByClass(Class<?> clazz) {
```

Comment:
Do we expect this to be called a bunch? The performance of this is pretty slow, so we may want to make the map bi-directional.

Reply:
yeah ideally we should make this bi-directional. i was making it a GVK->class map b/c class->GVK doesnt look like a frequent case i suppose.. i left a TODO comment here and will do this in a follow-up. wdyt?